main_test.go

```diff
@@ -42,3 +41,61 @@ func TestDirIsEmpty(t *testing.T) {
 		t.Fatalf(`Expected %s not to be empty, was.`, parent)
 	}
 }
+
+func TestRepoHasScheme(t *testing.T) {
+	testData := map[string]bool{
+		"git@github.com:some/thing.git":              false,
+		"http://what.dk/ever.git":                    true,
+		"https://github.com/something/different.git": true,
+	}
+
+	for addr, expected := range testData {
+		if hasScheme(addr) != expected {
+			t.Fatalf(`Expected scheme presence: %q for "%s", got %q.`, expected, addr, !expected)
```

Comment:
For table driven tests like this, it may be more appropriate to use `t.Errorf` so all tests are exercised and reported on even if one of them fails.